### DISCUSSION TITLE
issue: validation not working when testing components inside FormProvider

### DISCUSSION THREAD

**Discussion by jmcdl** [react-hook-form/react-hook-form#10593] [Q&A] [answered]
### Version Number

7.43.9

### Codesandbox/Expo snack

https://github.com/jmcdl/form-provider-tests

### Steps to reproduce

1. Create a `Form` component with one required input inside a child `Details` component,  using a `FormProvider`. 
2. Add a test for the validation on the `Form` component.
3. Add a test for the validation on the `Details` component, mocking the `FormProvider`.
4. Validation test passes correctly on `Form` component, but fails on `Details` component.

This issue came up on a project I'm working on where we have a large form component. I'd like to break the tests into separate files for each subsection because the test file is getting too long, but I haven't been able to figure out why the validation is not running, even though the `FormProvider` appears to be correctly wrapping the test components. 

This project is using vitest and react-testing-library for tests.

I have created a small demo repo of this with the tests, since codesandbox doesn't handle vitest tests very well. The repo is [here](https://github.com/jmcdl/form-provider-tests).

### Expected behaviour

Both tests should pass, since they're testing the same thing.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment, marked as answer]
Have a look here for examples: https://github.com/react-hook-form/react-hook-form/blob/master/src/__tests__/useFormContext.test.tsx

**Reply by jmcdl** [reply]
Thanks. After looking at your example there again (I did look before I swear), I have determined that the root cause of this problem was: I am an idiot. The proximate cause is that in my app I set the `mode` option in `useForm` to `onTouched`. I.e.
```typescript
const formMethods = useForm<FormValues>({ mode: 'onTouched'});
```

Obviously my attempt at testing validation on input blur was not going to work without adding that in the test as well.
